core/src/main/scala/com/pingcap/tispark/write/TiBatchWriteTable.scala

```diff
@@ -41,47 +42,35 @@ class TiBatchWriteTable(
     val options: TiDBOptions,
     val tiConf: TiConfiguration,
     @transient val tiDBJDBCClient: TiDBJDBCClient,
-    val isTiDBV4: Boolean)
+    val isTiDBV4: Boolean,
+    val tiTable: TableCommon)
     extends Serializable {
   private final val logger = LoggerFactory.getLogger(getClass.getName)
 
   import com.pingcap.tispark.write.TiBatchWrite._
+
   @transient private val tiSession = tiContext.tiSession
   // only fetch row format version once for each batch write process
   private val enableNewRowFormat: Boolean =
     if (isTiDBV4) tiDBJDBCClient.getRowFormatVersion == 2 else false
-  private var tiTableRef: TiTableReference = _
-  private var tiDBInfo: TiDBInfo = _
-  private var tiTableInfo: TiTableInfo = _
-  private var tableColSize: Int = _
-  private var colsMapInTiDB: Map[String, TiColumnInfo] = _
-  private var colsInDf: List[String] = _
-  private var uniqueIndices: Seq[TiIndexInfo] = _
-  private var handleCol: TiColumnInfo = _
+  private val tiTableRef: TiTableReference = options.getTiTableRef(tiConf)
+  private val tiDBInfo: TiDBInfo = tiSession.getCatalog.getDatabase(tiTableRef.databaseName)
+  private val tableColSize: Int = tiTable.getTableInfo.getColumns.size()
+  private val colsMapInTiDB: Map[String, TiColumnInfo] =
+    tiTable.getTableInfo.getColumns.asScala.map(col => col.getName -> col).toMap
+  private var colsInDf: List[String] = df.columns.toList.map(_.toLowerCase())
+  private val uniqueIndices: Seq[TiIndexInfo] =
+    tiTable.getTableInfo.getIndices.asScala.filter(index => index.isUnique)
+  private val handleCol: TiColumnInfo = tiTable.getTableInfo.getPKIsHandleColumn
   private var tableLocked: Boolean = false
   private var autoIncProvidedID: Boolean = false
   // isCommonHandle = true => clustered index
-  private var isCommonHandle: Boolean = _
+  private val isCommonHandle = tiTable.getTableInfo.isCommonHandle
   private var deltaCount: Long = 0
   private var modifyCount: Long = 0
   @transient private var persistedDFList: List[DataFrame] = Nil
   @transient private var persistedRDDList: List[RDD[_]] = Nil
 
-  tiTableRef = options.getTiTableRef(tiConf)
-  tiDBInfo = tiSession.getCatalog.getDatabase(tiTableRef.databaseName)
-  tiTableInfo = tiSession.getCatalog.getTable(tiTableRef.databaseName, tiTableRef.tableName)
```

Comment:
why not change it to `tiTableInfo=table.getTableInfo`. so you need not change when you use  tiTableInfo